lib/Controller/Library.php

```diff
@@ -2545,4 +2546,44 @@ private function getRemoteFileExtension($url)
 
         return $extension;
     }
+
+    /**
+     * This is called when video finishes uploading.
+     * Saves provided base64 image as an actual image to the library
+     *
+     * @param Request $request
+     * @param Response $response
+     *
+     * @return Response
+     * @throws InvalidArgumentException
+     */
+    public function videoThumbnail($request, $response)
+    {
+        $sanitizedParams = $this->getSanitizer($request->getParams());
+        $libraryLocation = $this->getConfig()->getSetting('LIBRARY_LOCATION');
+
+        $image = $request->getParam('image');
+        $mediaId = $sanitizedParams->getInt('mediaId');
```

Comment:
At this point, do we need to check permissions for the `mediaId`?

Maybe it could also be called "addThumbnail" so that it is more generic?